pkg/transport/queuestorage/queuestorage.go

```diff
@@ -14,34 +14,36 @@ import (
 	"github.com/tidwall/sjson"
 )
 
-var (
-	MaxFileSize int64 = 100 * 1024 * 1024 // 100MB
-	MaxRows     int64 = 1_000
-	MaxFileAge        = 1 * time.Hour
-)
-
 type QueueStorage struct {
 	queue   queue.QueueBackend
 	storage filestore.StorageBackend
 
 	DataDir string
 	Workers int
 
-	fws         map[string]*FileWriter
-	fwsMu       sync.Mutex
-	closedFiles chan FileEvent
+	fws          map[string]*FileWriter
+	fwsMu        sync.Mutex
+	closedFiles  chan FileWriterInfo
+	timeProvider func() time.Time
 
 	wg   sync.WaitGroup
 	done chan bool
 }
 
-func NewQueueStorageTransport(queue queue.QueueBackend, storage filestore.StorageBackend) *QueueStorage {
```

Comment:
I would prefer to keep the original function signature, as both a Queue and Storage are required to create this object.

Reply:
I changed the signature as a result of https://github.com/scratchdata/ScratchDB/pull/67#discussion_r1428250058 . Although, the usage of struct param was not entirely necessary.